From ebd9dc2a07ee3d56e0e29fd72d69af1a37c9e6ef Mon Sep 17 00:00:00 2001
From: "Juvenal A. Silva Jr" <juvenal@linux.vnet.ibm.com>
Date: Thu, 5 Dec 2013 15:43:28 -0200
Subject: [PATCH] Build change for PowerLinux compatibility

This patch changes the build system to take into account POWER arch specific
definitions, filtering information about endianess earlier for the build
---
 build/linux.gcc.inc |  6 ++++++
 build/linux.inc     | 31 ++++++++++++++++++++++---------
 2 files changed, 28 insertions(+), 9 deletions(-)

diff --git a/build/linux.gcc.inc b/build/linux.gcc.inc
index 2b9c405..66101a1 100644
--- a/build/linux.gcc.inc
+++ b/build/linux.gcc.inc
@@ -94,11 +94,17 @@ endif
 
 ifeq (ppc64,$(arch))
     CPLUS_FLAGS += -m64
+    ifndef gcc_at
+        CPLUS_FLAGS += -mcpu=$(ppc_cpu) -mtune=$(ppc_cpu) -D__powerpc64__
+    endif
     LIB_LINK_FLAGS += -m64
 endif
 
 ifeq (ppc32,$(arch))
     CPLUS_FLAGS += -m32
+    ifndef gcc_at
+        CPLUS_FLAGS += -mcpu=$(ppc_cpu) -mtune=$(ppc_cpu) -D__powerpc32__
+    endif
     LIB_LINK_FLAGS += -m32
 endif
 
diff --git a/build/linux.inc b/build/linux.inc
index 0a40a4b..9e58d65 100644
--- a/build/linux.inc
+++ b/build/linux.inc
@@ -42,6 +42,16 @@ ifeq (icc,$(compiler))
   endif
 endif
 
+ifndef runtime
+        export gcc_at=$(shell gcc --version | grep "Advance-Toolchain")
+        gcc_version = $(shell gcc -dumpversion)
+        os_version:=$(shell uname -r)
+        os_kernel_version:=$(shell uname -r | sed -e 's/-.*$$//')
+        export os_glibc_version_full:=$(shell getconf GNU_LIBC_VERSION | grep glibc | sed -e 's/^glibc //')
+        os_glibc_version:=$(shell echo "$(os_glibc_version_full)" | sed -e '2,$$d' -e 's/-.*$$//')
+        export runtime:=cc$(gcc_version)_libc$(os_glibc_version)_kernel$(os_kernel_version)
+endif
+
 ifndef arch
         uname_m:=$(shell uname -m)
         ifeq ($(uname_m),i686)
@@ -56,6 +66,18 @@ ifndef arch
         ifeq ($(uname_m),sparc64)
                 export arch:=sparc
         endif
+        ifeq ($(uname_m),ppc32)
+                export arch := ppc32
+                ifeq ($(gcc_at),)
+                        export ppc_cpu:=$(shell cat /proc/cpuinfo | grep cpu | head -n 1 | sed -e 's/^cpu.*: //' | sed -e 's/ .*$$//' | tr '[:upper:]' '[:lower:]' | cut -f1 -d" " -)
+                endif
+        endif
+        ifeq ($(uname_m),$(filter $(uname_u),ppc64 ppc64le))
+                export arch := ppc64
+                ifeq ($(gcc_at),)
+                        export ppc_cpu:=$(shell cat /proc/cpuinfo | grep cpu | head -n 1 | sed -e 's/^cpu.*: //' | sed -e 's/ .*$$//' | tr '[:upper:]' '[:lower:]' | cut -f1 -d" " -)
+                endif
+        endif
         ifeq ($(uname_m),armv71)
                 export arch :=armv7
         endif
@@ -64,15 +86,6 @@ ifndef arch
         endif
 endif
 
-ifndef runtime
-        gcc_version = $(shell gcc -dumpversion)
-        os_version:=$(shell uname -r)
-        os_kernel_version:=$(shell uname -r | sed -e 's/-.*$$//')
-        export os_glibc_version_full:=$(shell getconf GNU_LIBC_VERSION | grep glibc | sed -e 's/^glibc //')
-        os_glibc_version:=$(shell echo "$(os_glibc_version_full)" | sed -e '2,$$d' -e 's/-.*$$//')
-        export runtime:=cc$(gcc_version)_libc$(os_glibc_version)_kernel$(os_kernel_version)
-endif
-
 native_compiler := gcc
 export compiler ?= gcc
 debugger ?= gdb
-- 
2.1.0

